Lib/test/test_posix.py

```diff
@@ -1476,6 +1478,106 @@ def test_empty_file_actions(self):
         )
         self.assertEqual(os.waitpid(pid, 0), (pid, 0))
 
+    def test_resetids(self):
+        pid = posix.posix_spawn(
+            sys.executable,
+            [sys.executable, '-c', 'pass'],
+            os.environ,
+            resetids=True
+        )
+
+        self.assertEqual(os.waitpid(pid, 0), (pid, 0))
+
+
+    def test_resetids_wrong_type(self):
```

Comment:
I think we don't need so much similar tests. Needed only one test per error case. For example if the argument must be an iterable of integers in specified range, then test with a non-iterable, a list containing a non-integer, and a list containing an invalid integer. Your multiple tests test only the single kind of errors.